subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -250,10 +250,34 @@ static int vr_shm_configure(struct ipc_static_vrings *vr, const struct backend_c
 		return -ENOMEM;
 	}
 
+	/*
+	 * conf->shm_addr  +--------------+  vr->status_reg_addr
+	 *		   |    STATUS    |
+	 *		   +--------------+  vr->shm_addr
+	 *		   |              |
+	 *		   |              |
+	 *		   |   RX BUFS    |
+	 *		   |              |
+	 *		   |              |
+	 *		   +--------------+
+	 *		   |              |
+	 *		   |              |
+	 *		   |   TX BUFS    |
+	 *		   |              |
+	 *		   |              |
+	 *		   +--------------+  vr->rx_addr (aligned)
+	 *		   |   RX VRING   |
+	 *		   +--------------+  vr->tx_addr (aligned)
+	 *		   |   TX VRING   |
+	 *		   +--------------+
+	 */
+
 	vr->shm_addr = conf->shm_addr + VDEV_STATUS_SIZE;
-	vr->shm_size = shm_size(num_desc, conf->buffer_size) - VDEV_STATUS_SIZE;
+	vr->shm_size = shm_size(num_desc, conf->buffer_size);
 
-	vr->rx_addr = vr->shm_addr + VRING_COUNT * vq_ring_size(num_desc, conf->buffer_size);
+	vr->rx_addr = ROUND_UP(vr->shm_addr +
+			       VRING_COUNT * vq_ring_size(num_desc, conf->buffer_size),
+			       VRING_ALIGNMENT);
```

Comment:
The point here is that `vq_ring_size` should be aligned because `conf->buffer_size` should be already aligned. The buffer size is set into the DT and while we recommend in the DT binding for the alignment, we also check that at init time in: https://github.com/zephyrproject-rtos/zephyr/blob/d8661f151d4b95ccbbd91f9163351688a7ce1b4a/subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c#L768-L769